modules/hal_silabs/Kconfig

```diff
@@ -12,7 +12,7 @@ config ZEPHYR_HAL_SILABS_MODULE_BLOBS
 config HAS_SILABS_GECKO
 	bool
 	select HAS_CMSIS_CORE
-	depends on SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1
+	depends on SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1 || SOC_FAMILY_SILABS_S2 
```

Comment:
Normally, Gecko is only used by Series-1 while Series-2 should use SiSDK. Can you explain why you need this change?